source/aws_solutions/qnabot/cli/qnabot_cli_helper.py

```diff
@@ -0,0 +1,446 @@
+                str_file_contents=str_file_contents
+            )  # convert to JSON format (if input is JSON Lines format)
+
+        try:
+            os.makedirs(os.path.dirname(export_filename), exist_ok=True)  # create export directory if does not exist
+            with open(export_filename, "w", encoding="utf-8") as obj_file:  # open file in write mode
+                obj_file.write(str_file_contents)  # write to file
+            return_response = {
+                "export_directory": export_filename,
+                "status": "Downloaded",
+                "comments": "Check the export directory for the downloaded export.",
+                "error_code": "none",
+            }
+            return_response = json.dumps(return_response, indent=4)
+            return return_response
+        except OSError as err_exception:
+            return error_response(
+                error_code=err_exception.errno,
+                message=err_exception.strerror,
+                comments="There was an issue using: " + export_filename + " Check the path and try again.",
+                status="Error",
+                show_error=True,
+            )
+    except ClientError as err_exception:
+        # if object has not been modified (304) or the object is not available in S3 bucket yet (NoSuchKey)
+        if err_exception.response["Error"]["Code"] in ("304", "NoSuchKey"):
+            return error_response(
+                error_code=err_exception.response["Error"]["Code"],
+                message=err_exception.response["Error"]["Message"],
+                comments="Please note: Export processing may take longer to process depending on the "
+                + "number of questions, and size of the download file.",
+                status="Pending",
+                show_error=False,
+            )
+        return error_response(
+            error_code=err_exception.response["Error"]["Code"],
+            message=err_exception.response["Error"]["Message"],
+            comments="",
+            status="Error",
+            show_error=True,
+        )
+
+
+def get_import_status(bucket: str, source_filename: str, importdatetime: datetime):
+    """
+    Get a file from a S3 bucket
+    :param bucket: Bucket to Get file from
+    :param key: S3 object name
+    :param importdatetime: the date time of the request in UTC timezone
+    :return: response status from the contents of the import request file
+    """
+
+    try:
+        s3_client = get_service_client("s3")  # boto3.client('s3')
+        # get object only if the object has changed since last request
+        key = "status/" + os.path.basename(source_filename)
+        logger.debug(f"Getting import status for {bucket=} {key=}")
+        response = s3_client.get_object(Bucket=bucket, Key=key, IfModifiedSince=importdatetime)
+
+        obj_status_details = json.loads(response["Body"].read().decode("utf-8"))  # read object body
+
+        return_response = {
```

Comment:
Use of if/else within the a dictionary with many keys, makes these line less readable.  Especially, when the else part of previous key/value appears next to the next key/value that doesn't start with an if.